src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as PaginationService from './PaginationService';
+import PaginationContainer from './PaginationContainer';
+import PaginationButton from './PaginationButton';
+import PageButton from './PageButton';
+
+/**
+ * Pagination is a component to show pages numbers calculate dinamically.
+ */
+
+const Pagination = ({
+  page,
+  perPage,
+  total,
+  pagesToShow,
+  onChange,
+  nextLabel,
+  previousLabel
+}) => {
+  const PaginationButtonContainer = ({ value }) => {
+    const active = page === value;
+    return (
+      <PaginationButton active={active} value={value} onClick={onChange} />
+    );
+  };
+
+  PaginationButtonContainer.propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
+  };
+
+  const totalPages = PaginationService.calculatePages(total, perPage);
+
+  if (totalPages < 2) {
+    return null;
+  }
+
+  if (totalPages <= pagesToShow) {
+    return (
+      <PaginationContainer
+        page={page}
+        totalPages={totalPages}
+        onChange={onChange}
+        nextLabel={nextLabel}
+        previousLabel={previousLabel}
+      >
+        {PaginationService.createEmptyArrayFromNumber(totalPages).map(
+          (item, index) => (
+            <PaginationButtonContainer key={item} value={index + 1} />
+          )
+        )}
+      </PaginationContainer>
+    );
+  }
+  const previousValues = PaginationService.arrayOfPreviousValues(
+    page,
+    totalPages
+  );
+  const nextValues = PaginationService.arrayOfNextValues(page, totalPages);
+
+  const shouldHavePreviousDots = PaginationService.shouldHavePreviousDots(
```

Comment:
I would name this based on the state instead of the representation, e.g.:

```suggestion
  const hasOmittedPreviousPages = PaginationService.hasOmittedPages(
```

This also reflects `shouldHaveNextDots` and the function names in the `PaginationService`.